From e0349a8514409420c60d89ca12d2d562628cb29c Mon Sep 17 00:00:00 2001
From: Martin Kaiser <wireshark@kaiser.cx>
Date: Mon, 20 Oct 2014 22:37:51 +0200
Subject: [PATCH 1/5] tn5250: fix an endless loop, exit when offset is not
 incremented

Bug: 10596
Change-Id: I64509aa49fce4893dee4e7a74a560e09a98830f6
Reviewed-on: https://code.wireshark.org/review/4878
Reviewed-by: Martin Kaiser <wireshark@kaiser.cx>
Tested-by: Martin Kaiser <wireshark@kaiser.cx>
(cherry picked from commit bc2726578156f3608960fc65ce1f691639e6addc)
Reviewed-on: https://code.wireshark.org/review/4879
Reviewed-by: Evan Huus <eapache@gmail.com>
Reviewed-on: https://code.wireshark.org/review/5338
Reviewed-by: Michael Mann <mmann78@netscape.net>
---
 epan/dissectors/packet-tn5250.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/epan/dissectors/packet-tn5250.c b/epan/dissectors/packet-tn5250.c
index a9ec088..0d46841 100644
--- a/epan/dissectors/packet-tn5250.c
+++ b/epan/dissectors/packet-tn5250.c
@@ -4651,6 +4651,8 @@ dissect_write_structured_field(proto_tree *tn5250_tree, tvbuff_t *tvb, gint offs
           length = tvb_get_guint8(tvb,offset);
           proto_tree_add_item(tn5250_tree, hf_tn5250_length, tvb, offset,
                               1, ENC_BIG_ENDIAN);
+          if (length==0)
+            break;
           proto_tree_add_item(tn5250_tree, hf_tn5250_dpt_ec, tvb, offset,
                               length, ENC_EBCDIC|ENC_NA);
           offset += length;
-- 
2.1.3

